Does not compile using the latest nightly

```
james@archx1cv4r2 ➜  rtfm-syntax git:(master) cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading syn v0.13.11                                                       
 Downloading backtrace v0.3.8                                                   
 Downloading backtrace-sys v0.1.19                                              
 Downloading libc v0.2.41                                                       
   Compiling unicode-xid v0.0.4                                                 
   Compiling cc v1.0.15
   Compiling quote v0.3.15
   Compiling unicode-xid v0.1.0
   Compiling libc v0.2.41
   Compiling rustc-demangle v0.1.8
   Compiling cfg-if v0.1.3
   Compiling either v1.5.0
   Compiling synom v0.11.3
   Compiling proc-macro2 v0.3.8
   Compiling syn v0.11.11
error[E0433]: failed to resolve. Could not find `Op` in `proc_macro`
   --> /home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.3.8/src/unstable.rs:131:42
    |
131 |                 let mut op = proc_macro::Op::new(tt.op(), spacing);
    |                                          ^^ Could not find `Op` in `proc_macro`

error[E0433]: failed to resolve. Could not find `Term` in `proc_macro`
   --> /home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.3.8/src/unstable.rs:446:59
    |
446 |             Span::Nightly(s) => Term::Nightly(proc_macro::Term::new(string, s)),
    |                                                           ^^^^ Could not find `Term` in `proc_macro`

error[E0412]: cannot find type `Term` in module `proc_macro`
   --> /home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.3.8/src/unstable.rs:439:25
    |
439 |     Nightly(proc_macro::Term),
    |                         ^^^^ not found in `proc_macro`
help: possible candidates are found in other modules, you can import them into scope
    |
3   | use Term;
    |
3   | use imp::Term;
    |
3   | use stable::Term;
    |

error[E0412]: cannot find type `Term` in module `proc_macro`
   --> /home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.3.8/src/unstable.rs:473:44
    |
473 |     fn unwrap_nightly(self) -> proc_macro::Term {
    |                                            ^^^^ not found in `proc_macro`
help: possible candidates are found in other modules, you can import them into scope
    |
3   | use Term;
    |
3   | use imp::Term;
    |
3   | use stable::Term;
    |

error[E0599]: no variant named `Op` found for type `proc_macro::TokenTree` in the current scope
   --> /home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.3.8/src/unstable.rs:228:13
    |
228 |             proc_macro::TokenTree::Op(tt) => {
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `proc_macro::TokenTree`

error[E0599]: no variant named `Term` found for type `proc_macro::TokenTree` in the current scope
   --> /home/james/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-0.3.8/src/unstable.rs:237:13
    |
237 |             proc_macro::TokenTree::Term(s) => ::Term::_new(Term::Nightly(s)).into(),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant not found in `proc_macro::TokenTree`

error: aborting due to 6 previous errors

Some errors occurred: E0412, E0433, E0599.
For more information about an error, try `rustc --explain E0412`.
error: Could not compile `proc-macro2`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -10,7 +10,7 @@ repository = "https://github.com/japaric/rtfm-syntax"
 version = "0.3.3"
 
 [dependencies]
-quote = "0.5.1"
+quote = "0.6.3"
 failure = "0.1.1"
 
 [dependencies.either]
@@ -19,8 +19,8 @@ version = "1"
 
 [dependencies.proc-macro2]
 features = ["nightly"]
-version = "0.3.6"
+version = "0.4.6"
 
 [dependencies.syn]
 features = ["extra-traits", "full"]
-version = "0.13.1"
+version = "0.14.2"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -2,7 +2,7 @@
 
 #![deny(missing_docs)]
 #![deny(warnings)]
-#![feature(proc_macro)]
+#![feature(proc_macro_diagnostic)]
 #![feature(try_from)]
 
 extern crate either;

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -61,7 +61,8 @@ impl App {
 
         let mut outcome = Outcome::default();
         for kv in app.kvs.into_pairs().map(|pair| pair.into_value()) {
-            let key = kv.key.as_ref();
+            let key_string = kv.key.to_string();
+            let key = key_string.as_ref();
             let span = kv.key.span();
             match key {
                 "device" => if let Either::Left(path) = kv.value {
@@ -221,7 +222,7 @@ impl Idle {
         let mut outcome = Outcome::default();
         for kv in idle.kvs.into_pairs().map(|p| p.into_value()) {
             let span = kv.key.span();
-            match kv.key.as_ref() {
+            match kv.key.to_string().as_ref() {
                 "path" => if let Either::Left(path_) = kv.value {
                     if path.is_some() {
                         outcome.error_duplicate_key(span);
@@ -296,7 +297,8 @@ impl Task {
 
         let mut outcome = Outcome::default();
         for kv in task.kvs.into_pairs().map(|pair| pair.into_value()) {
-            let key = kv.key.as_ref();
+            let key_string = kv.key.to_string();
+            let key = key_string.as_ref();
             let span = kv.key.span();
             match key {
                 "enabled" => if let TaskValue::Bool(b) = kv.value {

